### PR TITLE
ADAPT-2285: WYSIWYG

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- Add WYSIWYG component with spacing top/bottom and light text options
- Add to richTextRenderer.js to support H2-H6, heading links with icons, mapping images and links properly.
- Support for WYSIWYG links for minimal Basic Cards with light text.
- Update gatsby-plugin-postcss to 4.8.0

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-49--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-no-hero/
2. Look at the main content WYSIWYG section
3. Check that all heading, and heading links look correct with appropriate icons/hocus states.
4. Scroll to the section under Heading Level 2 and inspect the different links - external, internal, asset, email. No weird attributes should show up in the HTML, e.g., linktype (this is a prop of the link object within the WYSIWYG).
5. Inspect the first image inside the WYSIWYG below the intro text is downsized and processed to 1200px and 60% jpg quality, and that it loads lazy.
6. Inspect the 2nd smaller image, check that this image doesn't get upsampled (stays as original uploaded size) and it has an alt text.
6. Go to https://deploy-preview-49--stanford-alumni.netlify.app/test/yvonne/test-section/
7. Scroll to the section with black background and look at below card with link in body text, WYSIWYG links should now be accessible (hocus color is white)
Screen Shot 2021-06-25 at 2 01 28 AM

8. Check that posters look the same as before
https://deploy-preview-49--stanford-alumni.netlify.app/test/poster-test/
9. Look at code

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
Added a few inline styles for the WYSIWYG (tested on a different page and changes not published on Netlify)
Screen Shot 2021-06-25 at 9 58 19 AM

**Comment by yvonnetangsu** [comment]
@sherakama Unrelated to this PR but just a FYI - Last night I unpublished the one global alert in SAA that was published so it doesn't show up on all the previews. After I did that, I got a build error, so I went back to publish 1 global alert. Here's the error below. Also, it would be nice if the unpublished global alert don't show up on all the basic pages but that's not urgent.

Screen Shot 2021-06-24 at 11 34 46 PM

File: src/utilities/richTextRenderer.js [hunk 1]
```diff
@@ -49,12 +98,26 @@ const RichTextRenderer = ({ wysiwyg, className }) => {
...
+              {children}
+            </Heading>
+          );
+        }
+
         return null;
       },
+      [NODE_IMAGE]: (children, { src, alt }) => (
+        <CardImage size="horizontal" filename={src} alt={alt} loading="lazy" />
+      ),
```
Comment (sherakama) [inline src/utilities/richTextRenderer.js:113]: Cool

File: src/utilities/richTextRenderer.js [hunk 2]
```diff
@@ -49,12 +98,26 @@ const RichTextRenderer = ({ wysiwyg, className }) => {
...
     },
+    defaultStringResolver: (str) => <p>{str}</p>,
```
Comment (sherakama) [inline src/utilities/richTextRenderer.js:115]: Why did you need to add this here? What was the original?

File: src/tailwind/plugins/components/wysiwyg/heading-link.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+module.exports = function () {
+  return function ({ addComponents }) {
+    const components = {
+      ".heading-link-icon": {
+        "h2, h3, h4, h5, h6": {
+          "> a": {
+            "&::after": {
+              display: "inline-block",
+              content: "''",
+              mask: "url(/images/arrow-right.svg)",
```
Comment (sherakama) [inline src/tailwind/plugins/components/wysiwyg/heading-link.js:14]: hmmm. Is there a way to import this file so that the image path stays relative to the output or is that a big ask? I haven't looked into JIT compiling options for this.

**Review by sherakama — CHANGES_REQUESTED** [review]
Please review the image path question.

File: src/utilities/richTextRenderer.js (hunk 2 above)
Comment (yvonnetangsu) [reply src/utilities/richTextRenderer.js:115]: Oh, this is a new feature that the rich text resolver just added. Sometimes if text is pasted without the paragraph tag, it will add one around it (so `<div class="su-wysiwyg"><p>bunch of text</p></div>` instead of just `<div class="su-wysiwyg">bunch of text</div>`)

File: src/tailwind/plugins/components/wysiwyg/heading-link.js (hunk 3 above)
Comment (yvonnetangsu) [reply src/tailwind/plugins/components/wysiwyg/heading-link.js:14]: Yeah I spent a lot of time on this last night. It seems putting it in static is the easiest way for now. I'll wrap up the Giving stuff and investigate the TW support for this.

**Comment by sherakama** [comment]
> @sherakama Unrelated to this PR but just a FYI - Last night I unpublished the one global alert in SAA that was published so it doesn't show up on all the previews. After I did that, I got a build error, so I went back to publish 1 global alert, and the error went away. Here's the error below. Also, it would be nice if the unpublished global alert don't show up on all the basic pages but that's not urgent.
> 
> Screen Shot 2021-06-24 at 11 34 46 PM

Hmm. That is not good. I'll look into it.

**Review by sherakama — APPROVED** [review]
GTG

File: src/tailwind/plugins/components/wysiwyg/heading-link.js (hunk 3 above)
Comment (sherakama) [reply src/tailwind/plugins/components/wysiwyg/heading-link.js:14]: Create a ticket for it and come back to it later.

File: src/utilities/richTextRenderer.js (hunk 2 above)
Comment (sherakama) [reply src/utilities/richTextRenderer.js:115]: Neat

File: src/tailwind/plugins/components/wysiwyg/heading-link.js (hunk 3 above)
Comment (yvonnetangsu) [reply src/tailwind/plugins/components/wysiwyg/heading-link.js:14]: Thanks. I did a 5 minute search, and it seems this might be the only way for now. It's how people do it for Gridsome and other SSG also apparently. I haven't found anything with TW support yet.
